Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs

```diff
@@ -363,34 +363,37 @@ public Application(IAppxManifestApplication manifestApp, UWP package)
                 BackgroundColor = manifestApp.GetStringValue("BackgroundColor");
                 Package = package;
 
-                DisplayName = ResourceFromPri(package.FullName, DisplayName);
-                Description = ResourceFromPri(package.FullName, Description);
+                DisplayName = ResourceFromPri(package.FullName, package.Name, DisplayName);
+                Description = ResourceFromPri(package.FullName, package.Name, Description);
                 LogoUri = LogoUriFromManifest(manifestApp);
                 LogoPath = LogoPathFromUri(LogoUri);
 
                 Enabled = true;
             }
 
-            internal string ResourceFromPri(string packageFullName, string resourceReference)
+            internal string ResourceFromPri(string packageFullName, string packageName, string resourceReference)
             {
                 const string prefix = "ms-resource:";
                 if (!string.IsNullOrWhiteSpace(resourceReference) && resourceReference.StartsWith(prefix))
                 {
-                    // magic comes from @talynone
-                    // https://github.com/talynone/Wox.Plugin.WindowsUniversalAppLauncher/blob/master/StoreAppLauncher/Helpers/NativeApiHelper.cs#L139-L153
                     string key = resourceReference.Substring(prefix.Length);
                     string parsed;
                     if (key.StartsWith("//"))
                     {
-                        parsed = prefix + key;
-                    }
-                    else if (key.StartsWith("/"))
-                    {
-                        parsed = prefix + "//" + key;
+                        parsed = $"{prefix}{key}";
                     }
                     else
                     {
-                        parsed = prefix + "///resources/" + key;
+                        if (!key.StartsWith("/"))
+                        {
+                            key = $"/{key}";
+                        }
+
+                        if (!key.ToLower().Contains("resources"))
+                        {
+                            key = $"/Resources{key}";
+                        }
+                        parsed = $"{prefix}//{packageName}{key}";
```

Comment:
Does the project have tests? It'd be interesting to create unit tests for these additions

Reply:
Done